scripts/xblock/xblock_counts.py

```diff
@@ -0,0 +1,378 @@
+        root_val = response_json[BLOCK_ROOT_KEY]
+        counts = response_json[BLOCKS_KEY][root_val][BLOCK_COUNTS_KEY]
+        return counts
+    return {}
+
+
+def _get_block_summary_totals(course_data):
+    """
+    Totals the xBlock types included in the course data and returns those counts by type
+
+    Arguments:
+        course_data (list of dicts): a list of course_data objects
+
+    Returns:
+        dict: containing the total number of blocks by type
+            {
+                <block_type>: <count>,
+                ...
+            }
+        dict: containing the total unique courses for a block type
+    """
+    block_summary_counts = {}
+    unique_course_counts = {}
+
+    for course in course_data:
+        block_counts = course.get(BLOCK_COUNTS_KEY)
+        for count_label, value in block_counts.items():
+            if value > 0:
+                if count_label in block_summary_counts:
+                    block_summary_counts[count_label] += value
+                    unique_course_counts[count_label] += 1
+                else:
+                    block_summary_counts[count_label] = value
+                    unique_course_counts[count_label] = 1
+
+    return block_summary_counts, unique_course_counts
+
+
+def write_block_summary_report(course_data):
+    """
+    Generate a CSV file containing a summary of the xBlock usage
+
+    Arguments:
+        course_data (list of dicts): a list of course_data objects
+
+    Returns:
+        Nothing
+    """
+    (block_summary_counts, unique_course_counts) = _get_block_summary_totals(course_data)
+
+    # Open and start writing the data into the CSV
+    with open('xblock_summary_counts.csv', 'wb') as csvfile:
+        summary_writer = csv.writer(csvfile, delimiter=',',
+                                    quotechar='|', quoting=csv.QUOTE_MINIMAL)
+        summary_writer.writerow(['XBLOCK_NAME', 'UNIQUE_COURSES', 'NUM_TOTAL_INSTANCES'])
+        for block_type, block_count in block_summary_counts.items():
+            summary_writer.writerow([block_type, str(unique_course_counts[block_type]), str(block_count)])
+        csvfile.close()
+
+
+def write_course_block_detail_report(course_data):
+    """
+    Generate a CSV file containing the detailed information about the xBlocks available per course
+
+    Arguments:
+        course_data (list of dicts): a list of course_data objects
+
+    Returns:
+        Nothing
+    """
+    with open('xblock_course_detail.csv', 'wb') as csvfile:
+        detail_writer = csv.writer(csvfile, delimiter=',',
+                                   quotechar='"', quoting=csv.QUOTE_ALL)
+        detail_writer.writerow(['XBLOCK_TYPE_NAME', 'COURSE_NAME', 'COURSE_ID', 'COURSE_START', 'COURSE_END', 'NUM_XBLOCK_INSTANCES'])
+        for course in course_data:
+            for block_type, count in course.get(BLOCK_COUNTS_KEY, []).items():
+                if count > 0:
+                    detail_writer.writerow([block_type,
```

Comment:
Same as above- I think that when a dict initializer spans multiple lines, each item goes on its own line with the `[` and `]` going on their own lines. So here `block_type` would just need to get bumped down to its own line.